src/FESpaces/Assemblers.jl

```diff
@@ -80,21 +83,28 @@ function assemble!(r,a::Assembler,cv::CellMatrix)
 end
 
 """
-Assembler that produces SparseMatrices from the SparseArrays package
+Assembler that produces SparseMatrices from the SparseArrays and SparseMatricesCSR packages
 """
-struct SparseMatrixAssembler{E} <: Assembler{SparseMatrixCSC{E,Int},Vector{E}}
+struct SparseMatrixAssembler{E,M} <: Assembler{AbstractSparseMatrix{E,Int},Vector{E}}
   testfesp::FESpace
   trialfesp::FESpace
+
+  function SparseMatrixAssembler(
+    ::Type{M},
+    testfesp::FESpace{D,Z,T},
+    trialfesp::FESpace{D,Z,T}) where {M<:AbstractSparseMatrix,D,Z,T}
+    E = eltype(T)
+    new{E,M}(testfesp,trialfesp) 
+  end
 end
 
 function SparseMatrixAssembler(test::FESpace{D,Z,T}, trial::FESpace{D,Z,T}) where {D,Z,T}
-  E = eltype(T)
-  SparseMatrixAssembler{E}(trial,test)
+  SparseMatrixAssembler(SparseMatrixCSC, trial,test)
 end
 
 function assemble(
-  this::SparseMatrixAssembler{E},
-  vals::Vararg{Tuple{<:CellVector,<:CellNumber}}) where E
+  this::SparseMatrixAssembler{E,M},
```

Comment:
This change seems not to be needed if you don't use `M` in the funciton

```
  this::SparseMatrixAssembler{E},
  vals::Vararg{Tuple{<:CellVector,<:CellNumber}}) where E
```
was ok

Reply:
solved in 53d940e